Network logging overhaul

jukkar opened the pull request:
This PR converts the CONFIG_NET_DEBUG_* and CONFIG_SYS_LOG_NET_LEVEL options to just simple CONFIG_NET_*_LOG_LEVEL setting. This enables user to have more fine grained debugging. The code uses the new logging subsystem for logging.

jukkar committed:
net: Convert core IP stack to use log levels

Instead of one global log level option and one on/off boolean
config option / module, this commit creates one log level option
for each module. This simplifies the logging as it is now possible
to enable different level of debugging output for each network
module individually.

The commit also converts the code to use the new logger
instead of the old sys_log.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
drivers: eth: Convert to use new logging

Convert the ethernet drivers to use the new logging system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
@nordic-krch If you have time, please review and check if it makes sense what I am doing when converting to new logger.

nashif commented on drivers/ethernet/eth_native_posix.c:
```diff
@@ -208,7 +207,7 @@ static int eth_send(struct net_if *iface, struct net_pkt *pkt)
 	update_gptp(iface, pkt, true);
 
-	SYS_LOG_DBG("Send pkt %p len %d", pkt, count);
+	NET_DBG("Send pkt %p len %d", pkt, count);
```
why is this required? Why not sue LOG_DBG directly?

jukkar replied:
I had some issues when using logger directly. I will revisit this in next version.

mike-scott commented:
Is this PR setting a precedent?  Will every subsystem in Zephyr eventually convert from using CONFIG_SYS_LOG_* options to CONFIG_*_LOG_*?  I feel like this PR changes the net subsystem from one set of snowflake configs.. to another set.

jukkar committed:
drivers: net: slip: Convert to use new logging

Convert the driver to use the new logging system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
drivers: net: loopback: Convert to use new logging

Convert the driver to use the new logging system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: echo_server: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert net_app to use log level

Use network log level in net_app library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert coap API to use log level

Use network log level in CoAP library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert config library to use log level

Use network log level in config library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert dns library to use log level

Use network log level in DNS library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert http library to use log level

Use network log level in HTTP library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert openthread library to use log level

Use network log level in OpenThread library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert sntp library to use log level

Use network log level in SNTP library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert socket library to use log level

Use network log level in BSD socket library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert websocket library to use log level

Use network log level in Websocket library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

nashif commented:
> Is this PR setting a precedent? Will every subsystem in Zephyr eventually convert from using CONFIG_SYS_LOG_* options to CONFIG_LOG_*? I feel like this PR changes the net subsystem from one set of snowflake configs.. to another set.

yes, we will move to the new logging subsystem, not  sure if much though has been put in migration from the old to the new, i.e. I would have been happy if we kept CONFIG_SYS_LOG and just changed the implementation beneath it but the issue is that there are more changes needed, it is not a 1:1 translation.

Transition path based on what we have now: keep sys_log working, move to LOG everywhere, obsolete sys_log later.

jukkar commented:
New version uploaded. I made the patches a bit more fine grained. @mike-scott I separated the lwm2m changes to two separate patches if you want to handle them separately or do not want them to be merged.

About the new logger, I made a trivial and simplified test with frdm-k64f and ping.

New logger but no debug options enabled:
```
ping -c 64 192.0.2.1
PING 192.0.2.1 (192.0.2.1) 56(84) bytes of data.
64 bytes from 192.0.2.1: icmp_seq=1 ttl=64 time=0.602 ms
64 bytes from 192.0.2.1: icmp_seq=2 ttl=64 time=0.257 ms
```

Then same thing with new logger but various networking debug options enabled, like core, icmpv4, context etc
```
ping -c 64 192.0.2.1
PING 192.0.2.1 (192.0.2.1) 56(84) bytes of data.
64 bytes from 192.0.2.1: icmp_seq=1 ttl=64 time=0.409 ms
64 bytes from 192.0.2.1: icmp_seq=2 ttl=64 time=0.226 ms
64 bytes from 192.0.2.1: icmp_seq=3 ttl=64 time=0.211 ms
```
so basically no difference in execution between these two.

Then a control run with old logging and same debug options enabled as in previous run.
```
ping -c 64 192.0.2.1
PING 192.0.2.1 (192.0.2.1) 56(84) bytes of data.
64 bytes from 192.0.2.1: icmp_seq=1 ttl=64 time=85.0 ms
64 bytes from 192.0.2.1: icmp_seq=2 ttl=64 time=55.3 ms
```

pizi-nordic commented on include/net/net_core.h:
```diff
@@ -33,23 +33,31 @@ extern "C" {
+
+LOG_MODULE_REGISTER();
+
+#define NET_DBG(fmt, ...) LOG_DBG("(%p): %s: " fmt, k_current_get(), \
```
What about "(%p): %s(): " ?

jukkar replied:
Omitting () from function name saves two characters without sacrificing any readability. I am not sure about the print format atm, this is what we currently print with sys_log, but of course we could change that when printing with logger.

pizi-nordic commented on include/net/net_ip.h:
```diff
@@ -229,7 +229,7 @@ enum net_addr_type {
 	NET_ADDR_OVERRIDABLE,
 };
 
-#if NET_LOG_ENABLED > 0
```
I think we should use LOG_* constants instead of numeric values.

pizi-nordic commented on include/net/net_pkt.h:
```diff
@@ -687,7 +687,7 @@ static inline void net_pkt_set_src_ipv6_addr(struct net_pkt *pkt)
 			    CONFIG_NET_BUF_USER_DATA_SIZE, NULL)
 
-#if defined(CONFIG_NET_DEBUG_NET_PKT)
+#if CONFIG_NET_LOG_LEVEL_NET_PKT > 3
```
I think we should use LOG_* constants instead of numeric values.

pfalcon replied:
And probably use >= comparison for clarity. ("Logging starts at the level X", not "logging starts after the level Y".)

pizi-nordic commented on include/net/net_core.h:
```diff
@@ -33,23 +33,31 @@ extern "C" {
-#ifndef NET_SYS_LOG_LEVEL
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_NET_LEVEL
+#if !defined(LOG_LEVEL)
+#if defined(NET_LOG_LEVEL)
```
Why we need NET_LOG_LEVEL? Could we just use per-module LOG_LEVEL?

jukkar replied:
This was done like this so that networking header files can have network specific checks.
For example if LOG_LEVEL is used, it might turn on things in network header files even if networking is disabled.

pizi-nordic reviewed (commented):
General comment:
Soon we will have a console, which will be fully integrated with logger (https://github.com/zephyrproject-rtos/zephyr/issues/8872). To fully utilize capabilities of new logging subsystem I suggest the following:

- Keep some hierarchy module names. The dynamic log management will allow you to do something like this "log enable net_tcp_*".

- If something could be instantiated multiple times, please consider using LOG_INST*. Then you will be able to filter logs from given instance(s).

mike-scott commented on subsys/net/lib/lwm2m/ipso_light_control.c:
```diff
@@ -10,9 +10,9 @@
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_LWM2M_LEVEL
-#include <logging/sys_log.h>
+#define LOG_MODULE_NAME ipso_light_control
+#define LOG_LEVEL CONFIG_SYS_LOG_LWM2M_LEVEL
```
Do we want to change the Kconfig naming here to NET_LOG_LWM2M_LEVEL to match the rest of the subsys/net/lib options?

jukkar replied:
Hmm, I think I will leave to you to decide. I will remove the lwm2m patches from this PR, you can re-use them if you wish.

mike-scott commented on samples/net/echo_server/src/echo-server.c:
```diff
@@ -6,11 +6,8 @@
-#define NET_LOG_ENABLED 1
-#endif
+#define LOG_MODULE_NAME echo_server
+#define NET_LOG_LEVEL 4
```
Why switch to a hard coded 4 here?

pfalcon replied:
> Why switch to a hard coded 4 here?

I guess it is one-to-one translation. Was:

> #define NET_SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG

now:

> #define NET_LOG_LEVEL 4

Except, as mentioned by @pizi-nordic, should used symbolic constants everywhere.

mike-scott commented on samples/net/echo_server/src/tcp.c:
```diff
@@ -6,11 +6,8 @@
-#define NET_LOG_ENABLED 1
-#endif
+#define LOG_MODULE_NAME echo_server_tcp
+#define NET_LOG_LEVEL LOG_LEVEL_DBG
```
Almost all of the samples are using SYS_LOG_LEVEL_DEBUG elsewhere in the PR.  Should we be using LOG_LEVEL_DBG (like here) instead?

pfalcon commented on include/logging/log.h:
```diff
@@ -280,6 +280,39 @@ int log_printk(const char *fmt, va_list ap);
 #endif /*CONFIG_LOG_RUNTIME_FILTERING*/
 
 #else /* LOG enabled for the module. */
+/* No logging, so set the macros to do nothing */
```
@jukkar, ping. This commit ("logging: Avoid compile error if debug prints are disabled") should not be part of this PR (should be its own PR, merged long ago, or maybe no longer needed).

pfalcon commented on subsys/net/Kconfig.hostname:
```diff
@@ -29,9 +29,10 @@ config NET_HOSTNAME_UNIQUE
-	default n
+config  NET_LOG_LEVEL_HOSTNAME
+	int "Debug level hostname configuration"
+	default 0
```
Why we have "default 0" here (and "default 1" in other places). Default level should be ERROR (if not WARNING) consistently, to make sure that by default, errors don't go unnoticed by user.

pfalcon commented:
> This PR converts the CONFIG_NET_DEBUG_* and CONFIG_SYS_LOG_NET_LEVEL ...

> This PR converts the networking also to use new logging system

And this is very well-known and well-discussed mistakes, which yet continues to happen - please don't pack too much stuff into one PR. I'm peering for half an hour into this, and don't understand much. Sure, after I'll peer into this for a couple hours more, I'll understand more than that, but why that should be required?

pfalcon commented:
> Conflicting files
include/logging/log.h
subsys/net/Kconfig.hostname

Please rebase, this can't be applied to master to try it.

pfalcon commented:
> This PR converts the CONFIG_NET_DEBUG_* and CONFIG_SYS_LOG_NET_LEVEL options to just simple CONFIG_NET_LOG_LEVEL_* setting. This enables user to have more fine grained debugging.

I guess this needs more detailed explanation, mentioning CONFIG_NET_LOG_LEVEL_MAX and CONFIG_NET_LOG_LEVEL_DEFAULT, and with specific examples: before it was <long enough description>, now instead you do <long enough description>.

pfalcon commented on subsys/net/lib/config/Kconfig:
```diff
@@ -55,10 +55,10 @@ config NET_APP_NEED_IPV4
-	default n
-	default y if NET_LOG_GLOBAL
+config NET_LOG_LEVEL_CONFIG
+	int "Debug level net config library"
```
Should be "Log level for net config library". Similarly in many other places.

pfalcon commented on samples/net/echo_server/src/echo-server.c:
```diff
@@ -131,6 +128,13 @@ void main(void)
 		start_udp();
 	}
 
+	while (1) {
```
This of course can't be in the code ready for merging.

pfalcon reviewed (dismissed):
At least following should be done:

0. Rebase.
1. Don't use values like 0,1,2,3,4, use symbolic logging levels instead.
2. Debug logging help texts should look well in English.

Also ideally, converting to new logger should be split into different PR from switching to "snowflake" debug config. But my guess that's not possible (well, practical) at this stage.

P.S. As the new logger is an impeding doom, I'm all for getting over it, but we can't push sloppy conversion in just to get over it.

nordic-krch commented:
@jukkar regarding
> I think we should use LOG_* constants instead of numeric values.

Isn't it because LOG_LEVEL_... defines are in log.h file which must be included by C file because it utilizing defines? If so then I will move those defines to other file (something like log_types.h).

nordic-krch commented:
@jukkar

nordic-krch commented:
@jukkar accident close/reopen sequence.

Once #8704 was merged you can clean up examples and remove explicit log processing and initialization and rely on implicit logger handling (option CONFIG_LOG_PROCESS_THREAD must be enabled)

jukkar commented on drivers/net/Kconfig:
```diff
@@ -57,20 +57,19 @@ config	SLIP_MTU
-	int
-	prompt "SLIP driver log level"
-	depends on SYS_LOG && SLIP
+config	SLIP_LOG_LEVEL
```
I briefly looked about this and it is not possible to do things like this
	default y if NET_LOG_LEVEL > 3
in Kconfig if we use choice fields.

nordic-krch replied:
@ulfalizer is it possible to have some kind of templates in kconfig. As you can see in above comment, ideally we should have common scheme for configuring logging in modules.

jukkar commented on drivers/net/slip.c:
```diff
@@ -63,7 +66,7 @@ struct slip_context {
 
-#if SYS_LOG_LEVEL >= SYS_LOG_LEVEL_DEBUG
+#if LOG_LEVEL >= LOG_LEVEL_DBG
 #if defined(CONFIG_SYS_LOG_SHOW_COLOR)
```
The hexdump routine in slip.c needs conversion to use logger hexdump. The color check will disappear when that conversion is done.

jukkar commented:
Rebased and addressed most of the comments.

jukkar committed:
samples: net: coap_server: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: dhcpv4_client: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: dns_resolve: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: eth_native_posix: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: gptp: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: http_client: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: http_server: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: ipv4_autoconf: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: irc_bot: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: mdns_responder: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: promiscuous: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: throughput_server: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: traffic_class: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: vlan: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: ws_console: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Update which converts more sample apps to use log level.

nordic-krch commented on subsys/net/lib/http/http_client.c:
```diff
@@ -167,7 +164,7 @@ int http_request(struct http_ctx *ctx, struct http_request *req, s32_t timeout,
 }
 
-#if defined(CONFIG_NET_DEBUG_HTTP)
+#if NET_LOG_LEVEL > 2
```
numbers can now be replaced by define from log.h

nordic-krch commented on subsys/net/lib/http/http_client.c:
```diff
@@ -185,7 +182,7 @@ static void sprint_addr(char *buf, int len,
 			      enum http_method method)
 {
-#if defined(CONFIG_NET_DEBUG_HTTP)
+#if NET_LOG_LEVEL > 2
```
numbers can now be replaced by define from log.h

nordic-krch commented on subsys/net/lib/http/http_client.c:
```diff
@@ -272,7 +269,7 @@ int http_client_send_req(struct http_ctx *ctx,
 static void print_header_field(size_t len, const char *str)
 {
-#if defined(CONFIG_NET_DEBUG_HTTP)
+#if NET_LOG_LEVEL > 2
```
numbers can now be replaced by define from log.h

nordic-krch commented on subsys/net/lib/http/http_client.c:
```diff
@@ -434,7 +431,7 @@ static int on_headers_complete(struct http_parser *parser)
 static int on_message_begin(struct http_parser *parser)
 {
-#if defined(CONFIG_NET_DEBUG_HTTP) && (CONFIG_SYS_LOG_NET_LEVEL > 2)
+#if NET_LOG_LEVEL > 2
```
numbers can now be replaced by define from log.h

nordic-krch commented on subsys/net/lib/http/http_server.c:
```diff
@@ -81,7 +78,7 @@ void http_server_conn_monitor(http_server_cb_t cb, void *user_data)
 const char * const http_state_str(enum http_state state)
 {
-#if defined(CONFIG_NET_DEBUG_HTTP)
+#if NET_LOG_LEVEL > 3
```
numbers can now be replaced by define from log.h (multiple locations)

nordic-krch commented on subsys/net/lib/openthread/platform/logging.c:
```diff
@@ -10,9 +10,9 @@
-#define SYS_LOG_LEVEL 4
-#include <logging/sys_log.h>
+#define LOG_MODULE_NAME openthread
+#define LOG_LEVEL 4
```
LOG_LEVEL_DBG

jukkar commented on drivers/net/slip.c:
```diff
@@ -209,9 +210,9 @@ static int slip_send(struct net_if *iface, struct net_pkt *pkt)
 			slip_writeb_esc(c);
 		}
 
-#if SYS_LOG_LEVEL >= SYS_LOG_LEVEL_DEBUG
```
Yes, that hexdump + other hexdump calls will be converted eventually. As there is lot of code to change, this is still WIP and not yet meant to be merged.

jukkar commented on include/net/net_core.h:
```diff
@@ -33,36 +33,29 @@ extern "C" {
 #include <logging/log.h>
-#include <logging/log_ctrl.h>
 
 LOG_MODULE_REGISTER();
```
The registration is only done if user has defined NET_LOG_LEVEL before including net_core.h, I do not see much issue here atm.

jukkar commented on include/net/net_ip.h:
```diff
@@ -229,7 +229,7 @@ enum net_addr_type {
 };
 
-#if NET_LOG_ENABLED > 0
+#if NET_LOG_LEVEL > 0
```
I will of course go through the code many times and will fix these eventually, there is just of it so it takes time.

jukkar commented on drivers/net/loopback.c:
```diff
@@ -11,9 +11,10 @@
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_NET_LOOPBACK_LEVEL
-#include <logging/sys_log.h>
+#define LOG_MODULE_NAME netlo
+#define LOG_LEVEL CONFIG_NET_LOG_LEVEL_LOOPBACK
```
Using the template looks very nice, I will convert the code to do that.

jukkar commented on drivers/net/slip.c:
```diff
@@ -449,14 +450,14 @@ static int slip_init(struct device *dev)
 	slip->rx = NULL;
 	slip->first = false;
 
 #if defined(CONFIG_SLIP_TAP) && defined(CONFIG_NET_IPV4)
```
I agree, it is just not always possible because of missing symbols. There is separate efforts to convert code to use IS_ENABLED()

jukkar commented on samples/net/coap_client/src/coap-client.c:
```diff
@@ -4,11 +4,8 @@
-#define NET_LOG_ENABLED 1
-#endif
+#define LOG_MODULE_NAME net_coap_client
+#define NET_LOG_LEVEL LOG_LEVEL_DBG
```
The samples are just samples, and in this case they are related to networking functionality. The code can be re-used in user application in which case the user would need to convert them to his/hers needs anyway in which case the net logging macros would be converted or simply removed. I do not see big issues here. I do not mind if someone converts these samples to not use NET_*() logging macros.

jukkar committed:
net: buf: Convert to new logging system

This commit replaces syslog with new logging subsystem in buf.c

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
drivers: ieee802154: Convert to use new logging

Convert the IEEE 802.15.4 drivers to use the new logging system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: ip: Split debug prints into smaller pieces

Currently logging subsystem supports quite small number of function
parameters. So split some long functions into smaller pieces.
Hopefully this is just a temporary patch and we can support more
parameters to logging macros.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: mgmt: Use correct printf modifier in debug print

Net event information debug print was using wrong printf modifiers.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: Use logger hexdump print macro

As the logger provides hexdump macro, use that instead of our own.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: sockets: Convert to use new logging

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: stats: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: telnet: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: ws_echo_server: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: lib: Convert mqtt library to use log level

Use network log level in mqtt library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
tests: net: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: zperf: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: syslog: Remove the sample application

The application won't work anymore as it uses the old syslog
logging and networking is now converted to use the new logger
so there is no need for this sample application.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: sntp: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: lldp: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: nats: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: rpl_border_router: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: rpl_node: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
usb: Fix net_hexdump_frags() parameters

One parameter was missing when calling net_hexdump_frags() which
caused compile error.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
New version uploaded according to comments.
Issues:
* Net USB based apps fail to link and need conversion to new logging
* lwm2m conversion would need to be done

jukkar committed:
samples: net: mqtt_publisher: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
usb: net: Convert to use new logging

Convert the USB networking to use the new logging system.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
New version:
* converted net-usb to use new logger
* converted the code to use Kconfig templates for log level settings

Missing:
* lwm2m conversion

ping @mike-scott You indicated earlier that you wished to do the lwm2m logging conversion yourself. If this is still the case then we could add those lwm2m patches to this PR so that `sanitycheck -t net` could pass.

finikorg commented on subsys/usb/class/netusb/function_eem.c:
```diff
@@ -117,23 +119,23 @@ static void eem_read_cb(u8_t ep, int size, void *priv)
-		SYS_LOG_DBG("hdr 0x%x, eem_size %d, size %d",
+		LOG_DBG("hdr 0x%x, eem_size %d, size %d",
 			    eem_hdr, eem_size, size);
 
```
same here

finikorg commented on subsys/usb/class/netusb/function_rndis.c:
```diff
@@ -180,20 +182,20 @@ static int parse_rndis_header(const u8_t *buffer, u32_t buf_len)
+	LOG_DBG("Parsing packet: len %u payload offset %u payload len %u",
+		len, sys_le32_to_cpu(hdr->payload_offset),
 		    sys_le32_to_cpu(hdr->payload_len));
 
```
and here

finikorg reviewed (approved):
LGTM with minor nitpicks

jukkar commented:
Rebased against master and fixed indent issues in usb-net files.

jukkar committed:
samples: net: echo-*: Add openthread tag for sanitychecker

Add "openthread" tag for sanitychecker when it is running
OpenThread specific tests. This way it is easier to run just
those few OpenThread specific tests.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: wpan_serial: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: wpanusb: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Various sanitychecker fixes.

dbkinder reviewed (approved):
+1 for .rst file changes

jukkar committed:
net: lwm2m: Convert to new logging system

Use new logging system instead of SYS_LOG.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
samples: net: lwm2m_client: Convert to new logging system

Use new logging system instead of SYS_LOG.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Added lwm2m conversion and openthread compilation fixes.

jukkar commented:
recheck

jukkar commented:
ping @nordic-krch @pfalcon @mike-scott 
I managed to get the code to a stage where all the network related sanitychecks pass, so please re-review this so we can start to think about merging this.

nordic-krch commented on drivers/ethernet/Kconfig:
```diff
@@ -8,19 +8,41 @@
-	  2 WARNING, write SYS_LOG_WRN in addition to previous level
-	  3 INFO, write SYS_LOG_INF in addition to previous levels
-	  4 DEBUG, write SYS_LOG_DBG in addition to previous levels
+config ETHERNET_LOG_LEVEL_OFF
```
now you can use template #9933

nordic-krch commented on drivers/ethernet/eth_enc28j60.c:
```diff
@@ -664,7 +666,7 @@ static void eth_enc28j60_iface_init(struct net_if *iface)
 	struct eth_enc28j60_runtime *context = dev->driver_data;
 
-	SYS_LOG_DBG("");
+	LOG_DBG("");
```
is this needed?

nordic-krch commented on drivers/ethernet/eth_mcux.c:
```diff
@@ -413,22 +415,22 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
-			    hdr->port_id.clk_id[7],
-			    ntohs(hdr->port_id.port_number),
-			    ptpTsData->sequenceId);
+		LOG_DBG("PTP packet: ver %d type %d len %d "
```
it must be split into couple of messages as up to 9 args is supported.

nordic-krch commented on drivers/ethernet/eth_mcux.c:
```diff
@@ -413,22 +415,22 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
 		ptpTsData->sequenceId = ntohs(hdr->sequence_id);
 
 #ifdef CONFIG_ETH_MCUX_PHY_EXTRA_DEBUG
```
this flag is suspicious. We have log severity levels so this seems redundant.

jukkar replied:
This one just prints way too much data thus the extra #ifdef. If only we would have log level > DBG :)

nordic-krch commented on drivers/ieee802154/Kconfig:
```diff
@@ -15,24 +15,41 @@ menuconfig IEEE802154
-	int "IEEE802154 driver log level"
-	depends on SYS_LOG
-	default 0
+choice
```
template can be used.

jukkar commented:
I wonder how I am suppose to use this system now. When using legacy shell with the new logging I get

```
subsys/shell/shell_log_backend.c: In function ‘msg_to_fifo’:
../zephyr/subsys/shell/shell_log_backend.c:59:13: error: ‘CONFIG_SHELL_MAX_LOG_MSG_BUFFERED’ undeclared (first use in this function); did you mean ‘CONFIG_NET_MAX_LOG_LEVEL’?
  if (cnt >= CONFIG_SHELL_MAX_LOG_MSG_BUFFERED) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
             CONFIG_NET_MAX_LOG_LEVEL
../zephyr/subsys/shell/shell_log_backend.c:59:13: note: each undeclared identifier is reported only once for each function it appears in
```

nordic-krch commented on subsys/usb/class/netusb/function_ecm.c:
```diff
@@ -58,17 +60,17 @@ static u8_t tx_buf[NETUSB_MTU], rx_buf[NETUSB_MTU];
 			     u8_t **data)
 {
-	SYS_LOG_DBG("");
+	LOG_DBG("");
```
it looks redundant

nordic-krch commented on subsys/usb/class/netusb/Kconfig:
```diff
@@ -87,7 +87,7 @@ endif # USB_DEVICE_NETWORK_RNDIS
 if USB_DEVICE_NETWORK
 
-config SYS_LOG_USB_DEVICE_NETWORK_DEBUG_LEVEL
+config USB_DEVICE_NETWORK_DEBUG_LEVEL
```
use kconfig template

jukkar committed:
net: app: No need to print anything if resolving is not possible

The code was trying to print peer IP address string but that
pointer could contain garbage. There is actually no need to print
anything in this case, the error code return is enough.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
This PR should be finally nearing merge stage, could @pfalcon @mike-scott and @nordic-krch recheck their -1 and possibly ack the PR?

nordic-krch commented:
@jukkar do you plan to respond to my review comments? I don't see any changes.

jukkar commented:
> do you plan to respond to my review comments? I don't see any changes.

I thought I did for most of your comments, I will recheck if I missed some comment.

mike-scott commented:
@jukkar I tested this today, and I have a question re: how the logging was setup for subsys/net/lib/lwm2m vs. CoAP, HTTP or DNS:

For CoAP:
```
module=COAP
module-dep=NET_LOG
module-str=Log level for CoAP
module-help=Enables CoAP debug messages.
source "subsys/net/Kconfig.template.log_config.net"
```
This results in a .config item: `CONFIG_COAP_LOG_LEVEL`

Vs. LwM2M which seems to use a different library-based template:
```
module=LWM2M
module-str=LWM2M library
source "subsys/logging/Kconfig.template.log_config"
```
Which results in several .config items:
```
# CONFIG_LWM2M_LOG_LEVEL_OFF is not set
# CONFIG_LWM2M_LOG_LEVEL_ERR is not set
# CONFIG_LWM2M_LOG_LEVEL_WRN is not set
# CONFIG_LWM2M_LOG_LEVEL_INF is not set
CONFIG_LWM2M_LOG_LEVEL_DBG=y
CONFIG_LWM2M_LOG_LEVEL=4
```

Do we want to move LwM2M to use the same template as CoAP, HTTP and DNS?

mike-scott commented:
Ah.. I see it doesn't depend on NET_LOG.. hrm.

jukkar commented:
> Do we want to move LwM2M to use the same template as CoAP, HTTP and DNS?

Originally you did not use NET_xxx() logging macros so I did not start to change that. I reworked the Kconfig template a bit for this version but forgot to check if lwm2m could use it as is, I will revisit lwm2m patches in next version.

jukkar committed:
samples: net: leds_demo: Convert to new logging system

Use new logging system with support for network log level.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented on drivers/ieee802154/ieee802154_mcr20a.c:
```diff
@@ -40,7 +42,7 @@
-#if (SYS_LOG_LEVEL == 4)
+#if (LOG_LEVEL == 4)
 /* Prevent timer overflow during SYS_LOG_* output */
 #define _MACACKWAITDURATION		(864 / 16 + 11625)
```
Yes, but I am not sure about the actual numbers behind the check so will leave this #if still here.

jukkar commented on subsys/net/lib/app/server.c:
```diff
@@ -41,7 +38,7 @@ static void new_client(struct net_context *net_ctx,
 	NET_INFO("Connection from %s (%p)",
```
I will check all these after #10123 is merged.

jukkar commented:
New version:

* lwm2m fixes
* missed comments and change requests fixed

jukkar commented:
recheck

jukkar commented:
Fixed merge conflict

jukkar commented:
Removed one extra compile error patch for native_posix as the issue was fixed already in upstream.

jukkar commented:
@nordic-krch I went through all your comments and hopefully I did not miss anything important this time. I kept the network specific Kconfig template file as I wanted to have proper dependencies there. If possible, please re-review.

@mike-scott If you are ok with the lwm2m changes, please +1 this PR.

mike-scott commented:
@jukkar Reviewed and test built/tested the LwM2M patches and I'm +1 to merge.  But wanted to point out that the debug logging for the sample is now tied to the LwM2M subsys logging config.  I'm not sure if I saw that before.  Is that correct?
```
#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
```

In other samples, it's hard-coded to the DBG log level (as was the LwM2M client originally)
```
#define NET_LOG_LEVEL LOG_LEVEL_DBG
```

jukkar commented:
> In other samples, it's hard-coded to the DBG log level (as was the LwM2M client originally)

Ok, I can force it to be LOG_LEVEL_DBG. I will send a new version that fixes this in lwm2m and I will also add support for the transient debug strings PR that was merged yesterday.

pfalcon commented on drivers/ethernet/Kconfig:
```diff
@@ -8,19 +8,13 @@
-	  3 INFO, write SYS_LOG_INF in addition to previous levels
-	  4 DEBUG, write SYS_LOG_DBG in addition to previous levels
+if NET_L2_ETHERNET
+module=ETHERNET
```
I think we should set syntactic conventions on how to specify template params right away, and I'd say that without spaces around "=" these don't look pretty.

Actually, we already have ` = ` in the codebase, e.g. subsys/shell/Kconfig, so let's follow that.

cc: @ulfalizer

jukkar replied:
Indeed I seem to have missed couple of these template changes in drivers. I will send a PR shortly for these.

jukkar committed:
net: Use log_strdup() when printing debug strings

As the debugging print calls are async, all the strings that might
be overwritten must use log_strdup() which will create a copy
of the printable string.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Added log_strdup() to relevant logging calls.

mike-scott commented:
@jukkar: thanks.  LGTM.

jukkar commented:
* Fixed sanity errors.
* Added spaces around `=` when sourcing Kconfig template file

jukkar commented:
ping @nordic-krch and you recheck and +1 if ok

nordic-krch reviewed (approved):
@jukkar  i've been on sick leave thus the delay. looks ok.

jukkar commented:
>  i've been on sick leave thus the delay. looks ok.

No worries, thanks for the review!

pfalcon commented:
> Added spaces around = when sourcing Kconfig template file

For reference, I don't see that being changed, e.g. not even where I made a comment on that https://github.com/zephyrproject-rtos/zephyr/pull/8816/files#r221893651 (but there're more occurrences). Let me submit a ticket to clarify the style we should use.
